### PR TITLE
feature: update workflow & golang to newest stable release 1.19

### PR DESCRIPTION
**Pull request by komandar**
This PR upgrades the golang version to the current available version (1.19) and the workflows action versions.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -7,21 +7,21 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        terraform_version: [ "1.0.x" ]
+        terraform_version: [ "1.2.x" ]
```
Comment (LKaemmerling) [inline .github/workflows/test.yml]: I would prefer to add all three latest supported versions so:
```suggestion
        terraform_version: [ "1.0.x", "1.1.x", "1.2.x" ]

**Review by LKaemmerling — CHANGES_REQUESTED** [review]
Hey @komandar,

thank you! I had a small suggestion. Also, the golangci-lint tests fail, can you maybe have a look there?

**Comment by komandar** [comment]
> Hey @komandar,
> 
> thank you! I had a small suggestion. Also, the golangci-lint tests fail, can you maybe have a look there?

Hi @LKaemmerling 
Thanks for your feedback. I have added the other Terraform versions. Unfortunately the test cases don't seem to work with the older versions. I don't have as much insight into your CI as you do. Do you have a spontaneous idea what the problem could be?

**Comment by LKaemmerling** [comment]
@komandar thank you! Good job!

**Pull request merged**